### PR TITLE
KAFKA-12690:Remove deprecated Producer#sendOffsetsToTransaction

### PR DESCRIPTION
**Pull request by Rancho-7**
Remove deprecated Producer#sendOffsetsToTransaction

JIRA:[issues.apache.org/jira/browse/KAFKA-12690](https://issues.apache.org/jira/browse/KAFKA-12690)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala [hunk 1]
```diff
@@ -76,12 +75,11 @@ class TransactionsBounceTest extends IntegrationTestHarness {
 
   override protected def brokerCount: Int = 4
 
-  @nowarn("cat=deprecation")
   @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
   @MethodSource(Array("getTestQuorumAndGroupProtocolParametersClassicGroupProtocolOnly_KAFKA_17961"))
   def testWithGroupId(quorum: String, groupProtocol: String): Unit = {
     testBrokerFailure((producer, groupId, consumer) =>
-      producer.sendOffsetsToTransaction(TestUtils.consumerPositions(consumer).asJava, groupId))
+      producer.sendOffsetsToTransaction(TestUtils.consumerPositions(consumer).asJava, consumer.groupMetadata()))
```
Comment (chia7712) [inline core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala]: Please remove this duplicate test case :)

File: core/src/test/scala/integration/kafka/api/TransactionsTest.scala [hunk 2]
```diff
@@ -301,12 +300,11 @@ class TransactionsTest extends IntegrationTestHarness {
     assertEquals(3L, second.offset)
   }
 
-  @nowarn("cat=deprecation")
   @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
   @MethodSource(Array("getTestQuorumAndGroupProtocolParametersClassicGroupProtocolOnly_KAFKA_17961"))
   def testSendOffsetsWithGroupId(quorum: String, groupProtocol: String): Unit = {
     sendOffset((producer, groupId, consumer) =>
-      producer.sendOffsetsToTransaction(TestUtils.consumerPositions(consumer).asJava, groupId))
+      producer.sendOffsetsToTransaction(TestUtils.consumerPositions(consumer).asJava, consumer.groupMetadata()))
```
Comment (chia7712) [inline core/src/test/scala/integration/kafka/api/TransactionsTest.scala]: Please remove this duplicate test case :)

**Review by chia7712 — COMMENTED** [review]
@Rancho-7 thanks for your patch. two small comments are left. PTAL

File: core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala (hunk 1 above)
Comment (Rancho-7) [reply core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala]: > Please remove this duplicate test case :)

Okay.

File: core/src/test/scala/integration/kafka/api/TransactionsTest.scala (hunk 2 above)
Comment (Rancho-7) [reply core/src/test/scala/integration/kafka/api/TransactionsTest.scala]: > Please remove this duplicate test case :)

Okay.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by dajac** [comment]
@chia7712 @Rancho-7 Could you please add a note to the upgrade notes about this removal. We usually list all the removed methods there with a brief explanation.

**Comment by chia7712** [comment]
> Could you please add a note to the upgrade notes about this removal. We usually list all the removed methods there with a brief explanation.

Sorry, I overlooked that. @Rancho-7, could you please make a minor update to [this line](https://github.com/apache/kafka/blob/trunk/docs/upgrade.html#L25)?

**Comment by Rancho-7** [comment]
> add a note to the upgrade notes

Sure.Sorry about that.

**Comment by Rancho-7** [comment]
> > Could you please add a note to the upgrade notes about this removal. We usually list all the removed methods there with a brief explanation.
> 
> Sorry, I overlooked that. @Rancho-7, could you please make a minor update to [this line](https://github.com/apache/kafka/blob/trunk/docs/upgrade.html#L25)?

Sure.Sorry about that.

**Pull request merged**